subprojects/packagefiles/mpdecimal/meson.build

```diff
@@ -202,66 +200,120 @@ if machine == 'auto'
-
-elif machine == 'full-coverage'
-  add_project_arguments('-DCONFIG_32', '-DANSI', language : 'c')
-  conf.set('MPD_HEADER_CONFIG', mpd_header_config_fullcov)
+  add_project_arguments('-DCONFIG_64', '-DANSI', language: 'c')
+  if host_machine.system() != 'windows'
+    add_project_arguments('-m64', language: ['c', 'cpp'])
+    add_project_link_arguments('-m64', language: ['c', 'cpp'])
+  endif
 
 elif machine == 'ppro'
-  add_project_arguments('-DCONFIG_32', '-DPPRO', language : 'c')
   conf.set('MPD_HEADER_CONFIG', mpd_header_config_32)
+  add_project_arguments('-DCONFIG_32', '-DPPRO', '-DASM', language: 'c')
   if host_machine.system() == 'windows'
-    add_project_arguments('/DMASM', language: 'c')
+    add_project_arguments('-DMASM', language: 'c')
+  else
+    add_project_arguments('-DASM', language: 'c')
+    add_project_arguments('-m32', language: ['c', 'cpp'])
+    add_project_link_arguments('-m32', language: ['c', 'cpp'])
+  endif
+
+  # Some versions of gcc miscompile inline asm:
+  # http://gcc.gnu.org/bugzilla/show_bug.cgi?id=46491
+  # http://gcc.gnu.org/ml/gcc/2010-11/msg00366.html
+  if cc.get_id() == 'gcc' and cc.run('''
+    __attribute__((noinline)) int
+    foo(int *p) {
+      int r;
+      asm ( "movl \$6, (%1)\n\t"
+            "xorl %0, %0\n\t"
+            : "=r" (r) : "r" (p) : "memory"
+      );
+      return r;
+    }
+    int main() {
+      int p = 8;
+      if ((foo(&p) ? : p) != 6)
+        return 1;
+      return 0;
+    }''',
+    args: ['-O2'],
+    name: 'gcc ipa-pure-const').returncode() != 0
+
+    add_project_arguments('-fno-ipa-pure-const', language: 'c')
   endif
 
 elif machine == 'ansi32'
-  add_project_arguments('-DCONFIG_32', '-DANSI', language : 'c')
   conf.set('MPD_HEADER_CONFIG', mpd_header_config_32)
+  add_project_arguments('-DCONFIG_32', '-DANSI', language: 'c')
+  if host_machine.system() != 'windows'
+    add_project_arguments('-m32', language: ['c', 'cpp'])
+    add_project_link_arguments('-m32', language: ['c', 'cpp'])
+  endif
 
 elif machine == 'ansi-legacy'
-  add_project_arguments('-DCONFIG_32', '-DANSI', '-DLEGACY_COMPILER', language : 'c')
-  conf.set('MPD_HEADER_CONFIG', mpd_header_config_32_legacy)
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_32_LEGACY)
+  add_project_arguments('-DCONFIG_32', '-DANSI', '-DLEGACY_COMPILER', language: 'c')
+  if host_machine.system() != 'windows'
+    add_project_arguments('-m32', language: ['c', 'cpp'])
+    add_project_link_arguments('-m32', language: ['c', 'cpp'])
+  endif
 
 elif machine == 'universal'
-  add_project_arguments('-DUNIVERSAL', language : 'c')
-  conf.set('MPD_HEADER_CONFIG', mpd_header_config_universal)
+  add_project_arguments('-DUNIVERSAL', language: 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_UNIVERSAL)
+  if host_machine.system() == 'windows'
+    error('machine "universal" not supported on Windows')
+  endif
+
+else
+  error('unknown machine type "{0}"', machine)
```

Comment:
This doesn't work in either python or Meson. You need to format it.

Reply:
The check is kind of silly as the values for the machine option are constrained. I just removed the `error()`.